python/lsst/jointcal/jointcal.py

```diff
@@ -610,11 +610,7 @@ def _do_load_refcat_and_fit(self, associations, defaultFilter, center, radius,
         add_measurement(self.job, 'jointcal.associated_%s_fittedStars' % name,
                         associations.fittedStarListSize())
 
-        applyColorterms = False if name == "Astrometry" else self.config.applyColorTerms
-        if name == "Astrometry":
-            referenceSelector = self.config.astrometryReferenceSelector
-        elif name == "Photometry":
-            referenceSelector = self.config.photometryReferenceSelector
+        applyColorterms = False if name.lower() == "astrometry" else self.config.applyColorTerms
```

Comment:
So this was just not bombing because of the capitalization in `name`?!  Also, should `refObjLoader` and `referenceSelector` both be required parameters to `_do_load_refcat_and_fit` since having them as `None` is not really an option.  Oh, and I don't see `referenceSelector` in the docstring's **Parameters** list.

Reply:
I think I'm going to leave mucking around with `validate` until after some progress has been made on planning the [new refcat flux system](https://community.lsst.org/t/reference-catalogs-camfluxes-and-colorterms/3578). I added the `self.doPhotometry`/`self.doAstrometry` checks since those didn't require any assumptions about the selector.

Yeah, the mis-capitalization was causing that clause to not trigger, and thus not explode. The person who wrote it (me) was dumb.

Good call on making those required parameters. I've made them required *named* parameters, using the new python3 syntax for that, and cleaned up the docstrings.

Reply:
While I must take serious issue with the 2nd sentence of the 2nd paragraph above, I accept you position on validation... 😉